### ISSUE TITLE
SigmodFocalLoss return NAN

### ISSUE THREAD

**Issue by xueyoo** [open-mmlab/mmcv#1488]
Hi Team
I am using mmcv.ops.SigmodFocalLoss to calcate the solov2 category loss, and there are two class, and target also has data, but the result is NAN
blew is the test code :

import torch.nn as nn
import torch.nn.functional as F
import torch
from mmcv.ops import SigmoidFocalLoss

if __name__ == '__main__':
    cate_pred = torch.load('cate_pred.pth')
    cate_label = torch.load('cate_label.pth')
    loss = SigmoidFocalLoss(gamma=2.0, alpha=0.25)
    t2 = loss(cate_pred, cate_label)  # the result is NaN

[cate.zip](https://github.com/open-mmlab/mmcv/files/7533623/cate.zip)

is there anybody get me some suggestions？
thanks

**Comment by zhouzaida** [comment]
Hi @xueyoo , the unexpected result was caused by `cate_pred[14835]`

```python
>>> import torch.nn as nn
>>> import torch.nn.functional as F
>>> import torch
>>> from mmcv.ops import SigmoidFocalLoss

>>> cate_pred = torch.load('cate_pred.pth')
>>> cate_label = torch.load('cate_label.pth')
>>> cate_pred[14835]
tensor([ 9.3906, -4.2344], device='cuda:0', dtype=torch.float16,
       grad_fn=<SelectBackward>)
```
The value `9.3906` is too large so that `max((T)1. - p, (T)FLT_MIN)` approximately equals to 0.

https://github.com/open-mmlab/mmcv/blob/add157cc73b05c7be3d9b63505e17dc2288a6d0f/mmcv/ops/csrc/common/cuda/sigmoid_focal_loss_cuda_kernel.cuh#L23-L29

**Comment by xueyoo** [comment]
thanks for your kindly help
